_posts/2019-04-06-trees-and-rectangles-reconciling-sql-and-recursive-programming-languages.md

```diff
@@ -0,0 +1,159 @@
+---
+layout: post
+title: "trees and rectangles: reconciling SQL and recursive programming languages"
+description: ""
+category:
+tags: [haskell, sql]
+---
+
+There's a contradiction at the heart of web development.
+
+Most structures from HTML through to JSON calls are fundamentally
+hierarchical. Relationships are expressed primarily through
+embedding, and this means we mostly deal with trees.
+
+On the other hand, traditional SQL databases talk about
```

Comment:
It might be worth an aside to reinforce the idea that the reason for "rectangles" is that they model mathematical relations, rather than, say, because they are easily represented as two-dimensional arrays or something.